src/sentry/services/hybrid_cloud/identity/impl.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from typing import Any, List
+
+from sentry.services.hybrid_cloud.identity import APIIdentity, APIIdentityProvider, IdentityService
+
+
+class DatabaseBackedIdentityService(IdentityService):
+    def close(self) -> None:
+        pass
+
+    def get_provider(
+        self,
+        provider_id: int | None = None,
+        provider_type: str | None = None,
+        provider_ext_id: str | None = None,
+    ) -> APIIdentityProvider | None:
+        from sentry.models.identity import IdentityProvider
+
+        # If an id is provided, use that -- otherwise, use the type and external_id
+        idp_kwargs: Any = (
+            {"id": provider_id}
+            if provider_id
+            else {"type": provider_type, "external_id": provider_ext_id}
+        )
+
+        idp = IdentityProvider.objects.filter(**idp_kwargs).first()
+
+        return self._serialize_identity_provider(idp) if idp else None
+
+    def get_identity(
+        self,
+        provider_id: int,
+        user_id: int | None = None,
+        identity_ext_id: str | None = None,
+    ) -> APIIdentity | None:
+        from sentry.models.identity import Identity
+
+        # If an user_id is provided, use that -- otherwise, use the external_id
+        identity_kwargs: Any = {"user_id": user_id} if user_id else {"external_id": identity_ext_id}
+
+        identity = Identity.objects.filter(**identity_kwargs, idp_id=provider_id).first()
+
+        return self._serialize_identity(identity) if identity else None
+
+    def get_user_identities_by_provider_type(
+        self,
```

Comment:
Minor nit, but I've been making a habit of forcing `kwds only` style method calls.  That is, `*` between `self` and arguments, and forcing callers to explicitly add arguments.

The reason being that many of these service interfaces need to change *ALOT* because we're still nailing things down.  When these services get shared into `getsentry` it becomes impossible to change the positional arguments due to the need to change any `getsentry` calls in a separate PR.  ensuring that arguments are kwds makes it more possible to add or adjust argument behavior without breaking other callsites.  But, this is only a minor nit to consider.

Reply:
I saw the `*,` when referencing your code but wasn't sure what it was doing. I like that pattern, and I don't mind making the change to follow suit :+1: